### PR TITLE
[14.0] [IMP] `account_invoice_margin`: several small improvements

### PR DESCRIPTION
**Pull request by ivantodorovich**
Easier to review commit by commit

### CODE REVIEW

File: account_invoice_margin/report/account_invoice_report.py [hunk 1]
```diff
@@ -7,8 +7,22 @@
 class AccountInvoiceReport(models.Model):
     _inherit = "account.invoice.report"
 
-    margin = fields.Float(string="Margin", readonly=True)
+    margin = fields.Float(readonly=True)
+    purchase_price = fields.Float("Cost", readonly=True)
```
Comment (legalsylvain) [inline account_invoice_margin/report/account_invoice_report.py:11]: naive question (not blocking) : ``margin`` is ``Monetary`` on ``account.move.(line)`` and ``Float`` on report. is it attended ?

**Review by legalsylvain — APPROVED** [review]
Thanks a lot for all that changes ! 

LGTM. Code review / no test.

File: account_invoice_margin/report/account_invoice_report.py (hunk 1 above)
Comment (ivantodorovich) [reply account_invoice_margin/report/account_invoice_report.py:11]: I tried to use Monetary here, but then odoo started complaining about not knowing the currency field 
As I couldn't find any Monetary fields on this report, I left it as Float 😅 I didn't dig much tbh

**Comment by ivantodorovich** [comment]
Just rebased

File: account_invoice_margin/models/account_move_line.py [hunk 2]
```diff
@@ -4,55 +4,6 @@
 from odoo import api, fields, models
 
 
-class AccountMove(models.Model):
```
Comment (pedrobaeza) [inline account_invoice_margin/models/account_move_line.py]: This change is not correct, as it was that way on purpose, as models coming from one2many are allowed and even recommended to be kept on the same file.

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#file-naming

File: account_invoice_margin/models/account_move_line.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def _get_purchase_price(self):
         self.ensure_one()
         return self.product_id.standard_price
 
-    @api.depends("product_id", "product_uom_id")
+    @api.depends("product_id", "product_uom_id", "move_id.move_type")
```
Comment (pedrobaeza) [inline account_invoice_margin/models/account_move_line.py]: Is it possible to change the type of an invoice "on the fly"?

File: account_invoice_margin/models/account_move_line.py (hunk 2 above)
Comment (ivantodorovich) [reply account_invoice_margin/models/account_move_line.py]: Oh, I always forget about that. My personal taste differs here.
Anyway, reverted.

File: account_invoice_margin/models/account_move_line.py (hunk 3 above)
Comment (ivantodorovich) [reply account_invoice_margin/models/account_move_line.py]: In odoo standard: action_switch_invoice_into_refund_credit_note
In any case, better to have the field dependencies. Even if you don't expect them to change, some other module might need it ^^

**Review by sergio-teruel — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Please fw-port it to v15 if needed.

**Pull request merged**